Interface Class Specifications/Display/InterfaceSpec-Display.md

```diff
@@ -0,0 +1,86 @@
+
+**From other classes**: Although not expected to be common, it may be possible to derive a display interface out of a set of other interfaces representing individual parameters.
+
+**To other classes**: If some display parameters are dynamic, and the hardware in question can read the value of the parameter at runtime as it changes, then it may be useful to expose that parameter separately as its own interface. For example, a display's center of projection might be exposed as an `OSVR_Vec2` interface.
+
+## Messages
+In general, to work smoothly with the rest of the OSVR system, distance units are meters and velocity units are meters/second.
+
+### Display
+#### Data
+- type (enum in fixed|hmd)
+- vendor (string)
+- model: (string)
+- product URI (uri-string RFC 3986)
+- driver URI (uri-string RFC 3986)
+- notes: string
+- number of DisplayInputs (whole number)
+- number of logical DisplaySurfaces (whole number)
+- k1[3] rgb radial distortion supported (boolean, hmd only?) and priority (whole number)
+- RGB radial distortion supported (boolean, hmd only?, supercedes k1[3] in most cases) and priority (whole number)
+- mono point mesh distortion supported (boolean, hmd only?) and priority (whole number)
+- RGB mesh distortion supported (boolean, hmd only?) and priority (whole number)
+
+#### Rationale
+The Display message is the root object of the Display interface. It references both the physical DisplayInputs and logical DisplaySurfaces, which define the mapping between them.
+
+### DisplayInput
+> one per number of display inputs in Display message.
+
+#### Data
+- resolution (width, height in whole numbers)
+- vendor PNPID (string)
+- product id (string)
+- platform-specific hardware instance path to uniquely identify it (??? string?)
+- if part of the desktop, origin on the desktop (X and Y in whole numbers)
+- On Windows, dx adapter number (??? string?)
+- logical id or platform-agnostic identifier (string? or int? referenced by logical topology)
+- location and orientation (physical orientation and location? relative to what?)
+- eye relief/lense-to-screen distance (if not represented with location/orientation above)
+- Scan-out direction (unit OSVR_Vec2)
+- Scan-out framerate (analog/float)
+- estimated latency to start of render in ms (analog/float)
+
+#### Rationale
+The DisplayInput message represents a physical display input.
+
+### DisplaySurface
+> one per number of logical display surfaces in Display message
+
+#### Data
+- logical id of display input for this surface (from DisplayInput message)
+- eye index (whole number)
+- viewer index (whole number)
+- viewport (relative to this surface's corresponding DisplayInput)
+- rotation (enum in landscape|portrait|landscape-flipped|portrait-flipped in clockwise order, relative to corresponding DisplayInput)
```

Comment:
OK, let's call it rotation instead of orientation, and rename the enum values to (none|ninety_cw|one_eighty_cw|two_seventy_cw) or something similar.

The DisplaySurface orientation represents the 2D rotation transformation needed to get from a "right side up" render target onto the DisplayInput viewport. No additional transformation is done (in 2D) after that (the DisplayInput could be thought of as the final "framebuffer" for that input). The DisplayInput orientation parameter is intended to be used along with the DisplaySurface "rotation" for calculating the projection for a given DisplaySurface.

As for a concrete example, let's say the HMD has two screens (two DisplayInputs) and they each have a slight inward tilt, which would be represented by the orientation parameter on the DisplayInput. The left screen is rotated 180 degrees. In the HMD's "native" mode, that would result in the DisplaySurface for the right DisplayInput being "none", and the one on the left being "one_eighty_cw". If the displays were not in their native modes, and instead in landscape mode (90 degrees cw from the native mode), then the right DisplaySurface would be rotation = "two_seventy_cw" and the left would be "ninety_cw".